test/TestUtilities/Test.Utility/TestServer/PortReserverOfMockServer.cs

```diff
@@ -10,17 +12,17 @@
 using System.Threading;
 using NuGet.Common;
 
-namespace NuGet.CommandLine.Test
+namespace NuGet.Test.Server
 {
     /// <summary>
-    /// This class allocates ports while ensuring that:
+    /// This class allocates ports for package signing tests while ensuring that:
     /// 1. Ports that are permanently taken (or taken for the duration of the test) are not being attempted to be used.
     /// 2. Ports are not shared across different tests (but you can allocate two different ports in the same test).
     /// 
     /// Gotcha: If another application grabs a port during the test, we have a race condition.
     /// </summary>
     [DebuggerDisplay("Port: {PortNumber}, Port count for this app domain: {_appDomainOwnedPorts.Count}")]
-    public class PortReserver : IDisposable
+    public class PortReserverOfMockServer : IDisposable
```

Comment:
I don't understand this rename.  There's no `MockServer` dependency here.  Why not just keep the name `PortReserver`?

Reply:
Since there are two different PortReserver.cs in out dev branch before this PR:
https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/TestServer/PortReserver.cs
https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/PortReserver.cs
The two PortReserver.cs have different APIs.
And Dotnet.Integration.Test in this PR needs a third one ( the same PortReserver with the second one).
So I move the second PortReserver into Test.Utility, but need to rename to avoid duplicate class name.

Reply:
I'll create another issue investigating if we could merge two PortResever, if it's necessary.
I agree with you that the naming is not good for now.